src/test/java/org/smartregister/domain/TaskTest.java

```diff
@@ -83,7 +85,70 @@ public void testLocationAndRequesterMethods() {
 		task.setRequester(user);
 		assertEquals(location, task.getLocation());
 		assertEquals(user, task.getRequester());
-
+		
 	}
-
+	
+	@Test
+	public void testGetTaskPriorityShouldReturnCorrectVariable() {
+		assertEquals(TaskPriority.ROUTINE, Task.TaskPriority.get("routine"));
+		assertEquals(TaskPriority.URGENT, Task.TaskPriority.get("urgent"));
+		assertEquals(TaskPriority.ASAP, Task.TaskPriority.get("asap"));
+		assertEquals(TaskPriority.STAT, Task.TaskPriority.get("stat"));
+	}
+	
+	@Test(expected = IllegalArgumentException.class)
+	public void testGetTaskPriorityShouldReturnErrorWhenNull() {
+		assertEquals(TaskPriority.ROUTINE, Task.TaskPriority.get(null));
+	}
+	
+	@Test(expected = IllegalArgumentException.class)
+	public void testGetTaskPriorityShouldReturnErrorForInvalidEnums() {
+		assertEquals(TaskPriority.ROUTINE, Task.TaskPriority.get("normal"));
+	}
+	
+	@Test
+	public void testRestriction() {
+		Task task = new Task();
+		assertNull(task.getRestriction());
+		Restriction restriction = new Restriction();
+		restriction.setRepetitions(1);
+		restriction.setPeriod(new Period(new DateTime(), null));
+		
+		task.setRestriction(restriction);
+		assertEquals(restriction, task.getRestriction());
+		
+	}
+	
+	@Test
+	public void testPriorityConversionusingOrdinalSerialization() {
+		Gson gson = new GsonBuilder().registerTypeAdapter(DateTime.class, new TaskDateTimeTypeConverter())
+		        .registerTypeAdapter(TaskPriority.class, new PriorityOrdinalConverter()).create();
+		
+		Task task = new Task();
+		task.setPriority(TaskPriority.URGENT);
+		assertEquals("{\"priority\":2,\"serverVersion\":0}", gson.toJson(task));
+	}
+	
+	@Test
+	public void testPriorityConversionusingOrdinalLegacyRoutine() {
+		Gson gson = new GsonBuilder().registerTypeAdapter(DateTime.class, new TaskDateTimeTypeConverter())
+		        .registerTypeAdapter(TaskPriority.class, new PriorityOrdinalConverter()).create();
+		
+		String json="{\"priority\":3}";
+		Task task = gson.fromJson(json, Task.class);
+		assertEquals(TaskPriority.ROUTINE, task.getPriority());
+	}
+	
+	@Test
+	public void testPriorityConversionusingOrdinal() {
+		Gson gson = new GsonBuilder().registerTypeAdapter(DateTime.class, new TaskDateTimeTypeConverter())
+		        .registerTypeAdapter(TaskPriority.class, new PriorityOrdinalConverter()).create();
+		
+		String json="{\"priority\":1}";
+		Task task = gson.fromJson(json, Task.class);
+		assertEquals(TaskPriority.ASAP, task.getPriority());
```

Comment:
do you think?We can use `1,2,3,4` instead of `0,1,2,3`. I think we might end up with `0` when the priority is not set...

What do you think?

Reply:
All tasks on version 1 were being created with a default of 3 on all systems. I have confirmed e.g on reveal stage that has tasks being generated by various sources all values have 3

Reply:
Cool